projects/storefrontapp-e2e-cypress/cypress/e2e/regression/my-account/update-password.e2e.cy.ts

```diff
@@ -87,5 +88,77 @@ describe('My Account - Update Password', () => {
         });
       }
     );
+
+    describe('update password by agent', { testIsolation: false }, () => {
+      isolateTests();
+      before(() => {
+        standardUser.registrationData.email = generateMail(
+          randomString(),
+          true
+        );
+        cy.requireLoggedIn(standardUser);
+        cy.visit('/');
+      });
+
+      it('should display server error if agent try to modify customer password', () => {
+        cy.log('--> Agent logging in');
+        checkout.visitHomePage('asm=true');
+        cy.get('cx-asm-main-ui').should('exist');
+        cy.get('cx-asm-main-ui').should('be.visible');
+        asm.agentLogin('asagent', 'pw4all');
+        cy.log('--> Starting customer emulation');
+        startCustomerEmulation(standardUser.registrationData);
+
+        cy.selectUserMenuOption({
+          option: 'Password',
+        });
+
+        alerts.getErrorAlert().should('not.exist');
+        cy.get('[formcontrolname="oldPassword"]').type(
+          standardUser.registrationData.password
+        );
+        cy.get('[formcontrolname="newPassword"]').type(
+          updatePassword.newPassword
+        );
+        cy.get('[formcontrolname="newPasswordConfirm"]').type(
+          updatePassword.newPassword
+        );
+        cy.get('cx-update-password button.btn-primary').click();
+        cy.url().should('contain', updatePassword.PAGE_URL_UPDATE_PASSWORD);
+        alerts.getErrorAlert().should('contain', 'Access is denied');
+      });
+
+      after(() => {
+        signOutUser();
+      });
+    });
   });
 });
+
+export function startCustomerEmulation(customer, b2b = false): void {
```

Comment:
Why do you not use the exist function `asm.tartCustomerEmulation()`?

Reply:
since existing startcustomeremulation use different customer model which is not consistent with current standardUser        model

Reply:
I see, How about defining a const to meet the exsit customer model. e.g:
```
const customer = {
    fullName: standardUser.registrationData.firstName  + ' ' +standardUser.registrationData.lastName,
    email: standardUser.registrationData.email,
  };
```

Reply:
Agree